File: ios/Classes/streaming/utils/ParsedTime.m
#import "ParsedTime.h"

@implementation ParsedTime;


- (ParsedTime*)initWithData:(id) data {
    self = [super init];
    if(self) {
        if (data == [NSNull null] || data == nil) {
            self.hour = -1;
            self.minute = -1;
            self.isValid = NO;
        } else {
            self.hour = data[@"hour"] != [NSNull null] ? [data[@"hour"] intValue] : -1;
            self.minute = data[@"minute"] != [NSNull null] ? [data[@"minute"] intValue] : -1;
            self.isValid = !(self.hour < 0 || self.minute < 0);
        }
        
    }
    return self;
}

@end

File: android/src/main/java/com/comscore/flutter/streaming/utils/ParsedTime.java
package com.comscore.flutter.streaming.utils;

import static com.comscore.flutter.Args.getArgValue;

public class ParsedTime {
    public final Integer hour, minute;
    public ParsedTime (Object data) {
        if (data == null) {
            hour = null;
            minute = null;
            return;
        }
        hour = getArgValue(data, "hour");
        minute = getArgValue(data, "minute");
    }

    public boolean isValid() {
        return hour != null && minute != null;
    }
}
